pkg/blockbuilder/scheduler/scheduler.go

```diff
@@ -78,14 +121,31 @@ func (s *BlockBuilderScheduler) running(ctx context.Context) error {
 	}
 }
 
+func (s *BlockBuilderScheduler) completeObservationMode() {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	if s.observationComplete {
+		return
+	}
+
+	s.jobs = newJobQueue(s.cfg.JobLeaseExpiry, s.logger)
+
+	if err := s.finalizeObservations(); err != nil {
+		level.Warn(s.logger).Log("msg", "failed to compute state from observations", "err", err)
+		// (what to do here?)
+	}
```

Comment:
Should this error be a terminal state for whole scheduler app, or maybe it can retry here?

Reply:
I suspect retrying won't change anything as it's some deterministic non-RPC code. It would really be a logic error that we'd want to fix post-haste. I think the scheduler should resume starting up and just log/increment metrics if this happens. Thanks for forcing me to think about it. :)